src/validation/global.rs

```diff
@@ -110,16 +114,23 @@ impl GlobalValidator {
                 }
             }
         }
-        //check enums
-        let duplication_enums =
-            index.get_global_qualified_enums().entries().filter(|(_, vars)| vars.len() > 1).map(
-                |(_, variables)| {
-                    (variables[0].get_qualified_name(), variables.iter().map(|v| &v.source_location))
-                },
-            );
 
-        for (name, locations) in duplication_enums {
-            self.report_name_conflict(name, &locations.collect::<Vec<_>>(), None);
+        // We create groups of "buckets", where the key of a group is the enum variant name and the
+        // value of the group is the bucket declared as a vector storing the enum variant Index entries.
+        // If the size of the bucket is >1, we know duplicates are present which we can subsequently report.
```

Comment:
Are there tests for this duplicate check? Considering the panic in the previous codegen test, I think it would make sense to add a validation test for duplicate enums aswell

Reply:
We had some, but none for enum names as far as I can tell. I've added two further tests, one for names and one for inline enum variants